### PR TITLE
Clean old builds on param value, fix shell check issues, default bot port

### PR DESCRIPTION
**Pull request by DanVanAtta**
- remove explicit bot param for port, set it to 8000 by default
- add parameters for restarting and cleaning old installations when new
software is installed. Set to true for preprod.
- in host_control, move single-use 'preprod-bot' variable to point of usage so that params can be defined in one spot.
- couple shell checks here and there with double quotes

---------

## Testing:

Of big note, the find and delete command is dangerous! Please have a second look around that to be sure execute that only on prerelease system. Below is some output from my testing of the find+delete:
```
admin@PreRelease:~$ BASE_FOLDER="/home/triplea/lobby"
admin@PreRelease:~$ export TAG=1.9.0.0.12378
admin@PreRelease:~$ find ${BASE_FOLDER} -mindepth 1 -maxdepth 1 -type d | grep -v $TAG
/home/triplea/lobby/1.9.0.0.12375
/home/triplea/lobby/1.9.0.0.12355


admin@PreRelease:~$ find ${BASE_FOLDER} -mindepth 1 -maxdepth 1 -type d 
/home/triplea/lobby/1.9.0.0.12375
/home/triplea/lobby/1.9.0.0.12378
/home/triplea/lobby/1.9.0.0.12355



admin@PreRelease:~$ find ${BASE_FOLDER} -mindepth 1 -maxdepth 1 -type d | grep -v $TAG | xargs rm -rf
rm: cannot remove '/home/triplea/lobby/1.9.0.0.12375/run_lobby.sh': Permission denied
rm: cannot remove '/home/triplea/lobby/1.9.0.0.12375/remove_lobby.sh': Permission denied
rm: cannot remove '/home/triplea/lobby/1.9.0.0.12375/.install.success': Permission denied
rm: cannot remove '/home/triplea/lobby/1.9.0.0.12375/config/lobby/lobby.properties': Permission denied
rm: cannot remove '/home/triplea/lobby/1.9.0.0.12375/version': Permission denied
rm: cannot remove '/home/triplea/lobby/1.9.0.0.12375/bin/triplea-lobby-1.9.0.0.12375-all.jar': Permission denied
rm: cannot remove '/home/triplea/lobby/1.9.0.0.12355/run_lobby.sh': Permission denied
rm: cannot remove '/home/triplea/lobby/1.9.0.0.12355/remove_lobby.sh': Permission denied
rm: cannot remove '/home/triplea/lobby/1.9.0.0.12355/.install.success': Permission denied
rm: cannot remove '/home/triplea/lobby/1.9.0.0.12355/config/lobby/lobby.properties': Permission denied
rm: cannot remove '/home/triplea/lobby/1.9.0.0.12355/version': Permission denied
rm: cannot remove '/home/triplea/lobby/1.9.0.0.12355/bin/triplea-lobby-1.9.0.0.12355-all.jar': Permission denied
```

### CODE REVIEW

File: roles/bot/bot.sh [hunk 1]
```diff
@@ -71,14 +80,23 @@ function disableOldBots() {
 }
 
 mkdir -p /home/triplea/bots/
-INSTALL_FOLDER=/home/triplea/bots/${TAG_NAME}
+BASE_FOLDER=/home/triplea/bots
+INSTALL_FOLDER=${INSTALL_FOLDER}/${TAG_NAME}
```
Comment (ssoloff) [inline roles/bot/bot.sh]: Should this be `INSTALL_FOLDER=${BASE_FOLDER}/${TAG_NAME}`?

File: roles/host_control.sh [hunk 2]
```diff
@@ -61,23 +51,23 @@ case "$(hostname)" in
...
     ${LOBBY} \
-       --tag-name ${LATEST_TAG} \
+       --auto-clean-old TRUE \
+       --auto-start TRUE \
+       --tag-name "${LATEST_TAG}" \
        --lobby-port ${PRE_PROD_LOBBY_PORT}
-    ${PRE_PROD_BOT} \
+   ${BOT} \
+      --auto-clean-old TRUE \
+      --auto-restart TRUE \
```
Comment (ssoloff) [inline roles/host_control.sh]: These switches don't match the ones defined in _bot.sh_ (`--clean-old-installs` and `--restart-on-new-version`).

File: roles/host_control.sh [hunk 3]
```diff
@@ -61,23 +51,23 @@ case "$(hostname)" in
...
-    if [ ! -d "/home/triplea/lobby/${LATEST_TAG}" ]; then
-      NEW_VERSION=1
-    fi
     ${LOBBY_DB} \
-       --tag-name ${LATEST_TAG}
+       --tag-name "${LATEST_TAG}"
     ${LOBBY} \
-       --tag-name ${LATEST_TAG} \
+       --auto-clean-old TRUE \
+       --auto-start TRUE \
```
Comment (ssoloff) [inline roles/host_control.sh]: These switches don't match the ones defined in _lobby.sh_ (`--clean-old-installs` and `--restart-on-new-version`).

**Comment by DanVanAtta** [comment]
Good catches @ssoloff , I'll work on fixing them. The initial pattern of testing was pretty effective, runnings scripts directly on systems to be modified, but not ideal in a number of ways. At some point investing in dev /local infrastructure to test updates more easily will make sense. In the meantime the diligent review is quite appreciated.

**Comment by DanVanAtta** [comment]
@ssoloff updated, looking pretty good to me now in self-review. Would you mind having a second look to double check this?

**Comment by ssoloff** [comment]
P.S. Maybe this has already been discussed, but I'm not sure why **Squash and merge** is not enabled for this repository.  It would make it much easier to track changes back to PRs and reduce the noise of multi-commit PRs like this one in the history.

**Comment by DanVanAtta** [comment]
re: merge strategy, it looks like we need different merge strategies when merging to master vs when we are merging to 'environment' branches (eg: prod). Squash-rebase of all changes on master to prod seems to go on to cause conflicts, so a rebase-merge seems the way to go when merging to prod. Otherwise squash seems appropriate when merging to master.

**Pull request merged**